index.js

```diff
@@ -21,14 +32,20 @@ module.exports = function (src, dest, opts, cb) {
 	var read = fs.createReadStream(src);
 	var readListener = onetime(startWrite);
 
-	read.on('error', cb);
+	read.on('error', function (err) {
+		if (err) {
+			err = new CpFileError('cannot read from `' + src + '`: ' + err.message, err);
```

Comment:
The `…: ' + err.message` part could also be done within `CpFileError`. Both approaches have their advantages and disadvantages.